### PR TITLE
Avoid maximum limit when mlflow saves information

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

Takeover https://github.com/optuna/optuna/pull/3530

## Description of the changes

- [x] Split information sent to MLflow into chunks of 100 or 1000 (as in the original PR
- [x] Add test

### CODE REVIEW

**Comment by xadrianzetx** [comment]
I've assigned myself as reviewer of #3530. Hope that's fine @nzw0301.

**Comment by nzw0301** [comment]
@xadrianzetx Thanks. It's perfect!

**Comment by xadrianzetx** [comment]
Cool. Will check it out over the weekend.

**Comment by HideakiImamura** [comment]
@contramundum53 Could you review this PR if you have time?

**Review by xadrianzetx — COMMENTED** [review]
Code looks great, but I was wondering if we (or anyone else using `mlflow` APIs) should care about implementation details such as batch sizes or max tag length. This makes us tightly coupled and might result in additional maintenance cost. Imo, ideally all this should be handled by `mlflow` internally, and at quick glance it seems like there are [some existing solutins](https://github.com/mlflow/mlflow/blob/be0ef13130ce52a4f3440cf883dfdda52481e1f0/mlflow/utils/autologging_utils/client.py#L86-L97).

Approach from this PR is fine for now, but it could be worth to reconsider later on. What do you think @nzw0301?

File: tests/integration_tests/test_mlflow.py
Comment (nzw0301) [inline tests/integration_tests/test_mlflow.py:569]: Sorry, this line should be removed.

**Comment by nzw0301** [comment]
Thank you for sharing your opinion and I agree with you; this might be able to be handled by mlflow's python API. So let me create a feature request issue on mlflow repo. If the issue sounds reasonable to them, I'll add a TODO comment to this PR (or as a follow-up).

**Review by xadrianzetx — APPROVED** [review]
Awesome! Seems like `mlflow` maintainers agree with your proposition, so I guess this one could be merged to unblock and removed later once changes are published `mlflow` side. LGTM!

**Pull request merged**
